src/sub_platforms/sql_opt/videx/videx_service.py

```diff
@@ -375,113 +427,122 @@ def before_request():
-
-
-@app.route('/videx/visualization/get_stats', methods=['GET'])
-def get_stats():
-    """
-    返回 videx_meta_singleton 当前的缓存大小。
-    """
-    non_task_cache: VidexTaskCache = videx_meta_singleton.non_task_cache
-    data_dict = {}
-    if non_task_cache is not None and non_task_cache.db_tasks_stats is not None:
-        data_dict = json.loads(non_task_cache.db_tasks_stats.to_json())
-    code, message, response_data = 200, "OK", {'stats': data_dict}
-    return jsonify(code=code, message=message, data=response_data)
-
-
-@app.route('/videx/visualization/status', methods=['GET'])
-def status():
-    """
-    返回 videx_meta_singleton 当前的缓存大小。
-    """
-    code, message, response_data = 200, "OK", {'cache': dict(videx_meta_singleton.cache)}
-    return jsonify(code=code, message=message, data=response_data)
+@ns.route('/create_task_meta')
+class CreateTaskMeta(Resource):
+    @ns.doc('Create Task Meta')
+    @ns.expect(task_meta_model)
+    @ns.response(200, 'Success', response_model)
+    @ns.response(400, 'Validation Error')
+    def post(self):
+        req_json_item = api.payload
+        global videx_meta_singleton
+        videx_meta_singleton.add_task_meta(req_json_item)
+    
+        code, message, response_data = 200, "OK", {}
+        return jsonify(code=code, message=message, data=response_data)
+
+
+@ns.route('/clear_cache')
+class ClearCache(Resource):
+    @ns.doc('Clear Cache')
+    @ns.expect(clear_cache_model)
+    @ns.response(200, 'Success', response_model)
+    def post(self):
+        global videx_meta_singleton
+        videx_meta_singleton.clear_cache({})
+
+        code, message, response_data = 200, "OK", {}
+        return jsonify(code=code, message=message, data=response_data)
+
+
+
+@ns.route('/update_gt_stats')
+class UpdateGTStats(Resource):
+    @ns.doc('Update GT Stats')
+    @ns.expect(update_gt_stats_model)
+    @ns.response(200, 'Success', response_model)
+    def post(self):
+        # 提供 gt 结果，用于测试 videx-py 的其他环节是否正确。这些 gt 可能由于算法无法完美贴合 innodb（多列 ndv、多列 rec_in_ranges），
+        # 也可能是由于新建索引后一些统计量变化。
+        #
+        # 传入格式为：
+        #     req_dict = {
+        #         "task_id": task_id,
+        #         "gt_stats_file": load_json_from_file(expect_meta_files[0]),
+        #         "gt_ndv_mulcol_file": load_json_from_file(expect_meta_files[3]),
+        #         "gt_rec_in_ranges": gt_rec_in_ranges,  # 可能为空
+        #         "gt_req_resp": gt_req_resp,
+        #     }
+        # 注意，收集最耗时的 "hist_file", "ndv_single_file" 反倒不会因为建删索引而变化，因此 update_gt_stats 不需要传入这些
+        #
+        # Returns:
+        #     创建成功与否
+        raise NotImplementedError
+
+
+@ns.route('/set_task_variables')
+class SetTaskVariables(Resource):
+    @ns.doc('set task variables')
+    @ns.response(200, 'Success', response_model)
+    def post(self):
+        # 主要是指定某个 task 是否启用 gt 数据
+        raise NotImplementedError
+
+
+@ns.route('/ask_videx')
+class AskVidEx(Resource):
```

Comment:
The lowercase version of "VidEx" looks a bit odd. Shall we consider changing it to "AskVidex"?
(I think three cases are acceptable in our project: "VIDEX", "videx", or "Videx")